test-e2e/config-import.js

```diff
@@ -0,0 +1,76 @@
+import test from 'node:test'
+import assert from 'node:assert/strict'
+import { createManager } from './utils.js'
+import { defaultConfigPath } from '../tests/helpers/default-config.js'
+
+test('config import - load default config when passed a path to `createProject`', async (t) => {
+  const manager = createManager('device0', t)
+  const project = await manager.getProject(
+    await manager.createProject({ configPath: defaultConfigPath })
+  )
+  const presets = await project.preset.getMany()
+  const fields = await project.field.getMany()
+  const translations = await project.$translation.dataType.getMany()
+  assert.equal(presets.length, 28, 'correct number of loaded presets')
+  assert.equal(fields.length, 11, 'correct number of loaded fields')
+  assert.equal(
+    translations.length,
+    870,
+    'correct number of loaded translations'
+  )
+})
+
+test('config import - load and re-load config manually', async (t) => {
+  const manager = createManager('device0', t)
+  const project = await manager.getProject(await manager.createProject())
+
+  const warnings = await project.importConfig({ configPath: defaultConfigPath })
+  let presets = await project.preset.getMany()
+  let fields = await project.field.getMany()
+  let translations = await project.$translation.dataType.getMany()
+
+  assert.equal(
+    warnings.length,
+    0,
+    'no warnings when manually loading default config'
+  )
+  assert.equal(presets.length, 28, 'correct number of manually loaded presets')
+  assert.equal(fields.length, 11, 'correct number of manually loaded fields')
+  assert.equal(
+    translations.length,
+    870,
+    'correct number of manually loaded translations'
+  )
+
+  // re load the config
+  await project.importConfig({ configPath: defaultConfigPath })
+  presets = await project.preset.getMany()
+  fields = await project.field.getMany()
+  translations = await project.$translation.dataType.getMany()
+  assert.equal(
+    presets.length,
+    28,
+    're-loading the same config leads to the same number of presets (since they are deleted)'
+  )
+  assert.equal(
+    fields.length,
+    11,
+    're-loading the same config leads to the same number of fields (since they are deleted)'
+  )
+  assert.equal(
+    translations.length,
+    870,
+    're-loading the same config leads to the same number of translations (since they are deleted)'
+  )
+})
+
+test('failing on loading multiple configs in parallel', async (t) => {
+  const manager = createManager('device0', t)
+  const project = await manager.getProject(await manager.createProject())
+  await assert.rejects(async () => {
+    await Promise.all([
+      project.importConfig({ configPath: defaultConfigPath }),
+      project.importConfig({ configPath: defaultConfigPath }),
+    ])
+  }, 'loading configs in parallell should throw')
```

Comment:
I think this will fix it:

```suggestion
  const results = await Promise.allSettled([
    project.importConfig({ configPath: defaultConfigPath }),
    project.importConfig({ configPath: defaultConfigPath }),
  ])
  assert.equal(results[0]?.status, 'fulfilled', 'first import should work')
  assert.equal(results[1]?.status, 'rejected', 'second import should fail')
```

The important change was switching `Promise.all` to `Promise.allSettled`. With `Promise.all`, the following would happen:

1. `Promise.all` is called.
1. The first import starts.
1. The second import starts, and fails immediately.
1. The result of `Promise.all` (from step 1) rejects, causing the test to end, and the database to be closed.
1. The first import continues and tries to write to the database, which fails because it's closed.

Changing to `Promise.allSettled` lets both finish before shutting everything down.

Reply:
owww, nice. That did it. Thanks for maintaining this loong back and forth! I'm gonna merge this